tests/cli/workflow/taxprofiler/test_cli_taxprofiler_run.py

```diff
@@ -0,0 +1,74 @@
+"""This script tests the run cli command"""
+import logging
+
+import pytest
+from _pytest.logging import LogCaptureFixture
+from click.testing import CliRunner
+
+from cg.cli.workflow.taxprofiler.base import run
+from cg.constants import EXIT_SUCCESS
+from cg.models.cg_config import CGConfig
+
+
+def test_without_options(cli_runner: CliRunner, taxprofiler_context: CGConfig):
+    """Test command without case_id argument."""
+    # GIVEN no case_id
+    # WHEN dry running without anything specified
+    result = cli_runner.invoke(run, obj=taxprofiler_context)
+    # THEN command should NOT execute successfully
+    assert result.exit_code != EXIT_SUCCESS
+    # THEN command should mention argument
+    assert "Missing argument" in result.output
+
+
+def test_with_missing_case(
+    cli_runner: CliRunner,
+    taxprofiler_context: CGConfig,
+    caplog: LogCaptureFixture,
+    case_id_does_not_exist: str,
+):
+    """Test run command with invalid case."""
+    caplog.set_level(logging.ERROR)
+    # GIVEN case_id not in database
+    assert not taxprofiler_context.status_db.get_case_by_internal_id(
+        internal_id=case_id_does_not_exist
+    )
+    # WHEN running
+    result = cli_runner.invoke(run, [case_id_does_not_exist], obj=taxprofiler_context)
+    # THEN command should NOT successfully call the command it creates
+    assert result.exit_code != EXIT_SUCCESS
+    # THEN ERROR log should be printed containing invalid case_id
+    assert case_id_does_not_exist in caplog.text
+    assert "could not be found" in caplog.text
+
+
+def test_with_config_use_nextflow(
+    cli_runner: CliRunner,
+    taxprofiler_context: CGConfig,
+    caplog: LogCaptureFixture,
+    taxprofiler_case_id: str,
+    taxprofiler_config,
+):
+    """Test command with case_id and config file using nextflow."""
+    caplog.set_level(logging.INFO)
+    # GIVEN case-id
+    case_id: str = taxprofiler_case_id
+
+    # GIVEN a mocked config
+
+    # WHEN dry running with dry specified
+    result = cli_runner.invoke(
+        run, [case_id, "--dry-run", "--use-nextflow"], obj=taxprofiler_context
+    )
+
+    # THEN command should execute successfully
+    assert result.exit_code == EXIT_SUCCESS
+
+    # THEN command should use nextflow
+    assert "using nextflow" in caplog.text
+    assert "path/to/bin/nextflow" in caplog.text
+    assert "-work-dir" in caplog.text
+    assert "-params-file" in caplog.text
```

Comment:
It would be nice to use a for loop for these asserts

Reply:
Do you mean something like:

```
 assertions = {
        "using nextflow": "Assertion failed: 'using nextflow' not found in caplog.text",
        "path/to/bin/nextflow": "Assertion failed: 'path/to/bin/nextflow' not found in caplog.text",
        "-work-dir": "Assertion failed: '-work-dir' not found in caplog.text",
        "-params-file": "Assertion failed: '-params-file' not found in caplog.text",
        "-resume": "Assertion failed: '-resume' not found in caplog.text",
    }

    for text, error_message in assertions.items():
        assert text in caplog.text, error_message
```

Reply:
More like:
```
for message in ["using nextflow", etc]:
  assert message in caplog.text
```

Reply:
Done